### PR TITLE
remove dev; update scripts; check builds; minor lints as I poked around

### PR DESCRIPTION
**Pull request by ghost**
## Type of change

- [ ] Bug Fix
- [X] New Feature
- [ ] Refactor
- [X] Project Build
- [x] Project Configuration

## Submission Checklist

- [ ] Have you followed the guidelines in our [Contributing document](https://github.com/fanappics/buoy/blob/master/CONTRIBUTING.md)?
- [ ] Did you create appropriate tests?

## Description

This removes old Playground/Dev server in favor of Styleguidist dev server. The idea is that we can use styleguide to develop the new components. Styleguide is our current "deliverable" so this will provide dev-prod parity. This will also encourage rebuild of styleguide.

## Related Issues / Bugs / Feature Requests

https://jira.fanatics.com/browse/CPDT-186

## Test Deployment URL

**NB** The best way to test this is to pull the branch locally and run `npm run dev` and check the output. Then realize that you can use `npm run styleguide` instead and use `/dev/` route to see the development/hot-reload/devtool version

https://rawgit.com/fanappics/buoy/ft-CPDT-186/docs/index.html

## Deployment Notes

- `npm run dev` is deprecated but not removed
- `npm run styleguide` will run a prod version of Styleguide at `/` and dev-friendly version on `/dev/`
- `npm run styleguide:build` will wipe the `docs` folder to make sure everything is refreshed

## Steps to Test or Reproduce
- Pull down the branch
- run `npm run styleguide` and access `http://localhost:6060/dev/`
- run `npm run styleguide:build`
- try to create a new component and see it show up in the styleguide
- Try editing components while the styleguide server is running:
hotreload

- Or Vue DevTools:
screen shot 2018-02-08 at 7 46 46 pm

- or sourcemaps (NB: you can even refresh the page to see styles update):
screen shot 2018-02-08 at 7 46 21 pm

### CODE REVIEW

File: build/config.js [hunk 1]
```diff
@@ -69,8 +69,8 @@ function genConfig (opts) {
       }),
       new webpack.BannerPlugin({
         banner: `/*!
-* buoy v${version}
-* Forged by John Leider
+* Buoy v${version}
+* by Fanatics, Inc.
```
Comment (kierachell) [inline build/config.js]: Dunno who that John guy is so I did a drive-by fix

File: build/styleguide.config.js [hunk 2]
```diff
@@ -15,8 +15,10 @@ module.exports = {
     require('./webpack.prod.config.js'),{
       devServer: {
 	contentBase: resolve('../docs'),
-	publicPath: '/docs/',
-	disableHostCheck: true
+	publicPath: '/dev/',
+	disableHostCheck: true,
+	openPage: '/dev/',
```
Comment (kierachell) [inline build/styleguide.config.js]: So this allows us to run prod and dev version of  Styleguide at the same time:

prod version is on `/`
dev version is on `/dev/`

File: package.json [hunk 3]
```diff
@@ -9,12 +9,11 @@
   },
   "scripts": {
     "build": "npm run build:dist && npm run build:es5",
-    "build:dev": "cross-env NODE_ENV=production node build/webpack.dev.config.js",
     "build:dist": "rimraf dist && cross-env NODE_ENV=production webpack --config build/config.js --progress --hide-modules",
     "build:es5": "rimraf es5 && cross-env NODE_ENV=es5 babel src --out-dir es5",
     "debug-build": "node --inspect --debug-brk build/config.js",
     "debug:test": "./node_modules/.bin/cross-env NODE_ENV=test node --inspect --inspect-brk ./node_modules/jest/bin/jest.js --no-cache --runInBand --verbose",
-    "dev": "cross-env NODE_ENV=development webpack-dev-server --config build/webpack.dev.config.js --open --hot",
+    "dev": "echo 'Use `npm run styleguide` and view the development version at http://localhost:6060/dev/\nYou can see individual componends in http://localhost:6060/dev/#!/<component-name>'; exit 0",
```
Comment (kierachell) [inline package.json]: Deprecation warning 😄

File: package.json [hunk 4]
```diff
@@ -24,8 +23,7 @@
     "prepare": "npm run build",
     "prepush": "./prepush.sh",
     "styleguide": "vue-styleguidist server --config ./build/styleguide.config.js",
-    "styleguide:build": "vue-styleguidist build --config ./build/styleguide.config.js",
-    "watch": "cross-env TARGET=development webpack --config build/config.js --progress --hide-modules --watch"
+    "styleguide:build": "rimraf docs && vue-styleguidist build --config ./build/styleguide.config.js"
```
Comment (kierachell) [inline package.json]: added total folder write just to avoid weird merge conflicts

**Review by thedevelopnik — APPROVED** [review]

**Review by elemjay19 — APPROVED** [review]
